src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -811,7 +810,7 @@ protected override void StartDeletedResource(ODataDeletedResource resource)
             {
                 // Writing a deleted resource within an entity set
                 DeltaResourceSetScope deltaResourceSetScope = this.ParentScope as DeltaResourceSetScope;
-                Debug.Assert(deltaResourceSetScope != null, "Writing child of delta set and parent scope is not DeltaResourceSetScope");
+                Debug.Assert(this.ParentScope.State == WriterState.Start || deltaResourceSetScope != null, "Writing child of delta set and parent scope is not DeltaResourceSetScope");

```

Comment:
shall we also update the error sentence to refresh the changes?

Reply:
I think the error text is still valid (and even more correct than previously).  Previously the assert only checked that the parent scope was a delta resource set scope, not that what was being written was being written as the child of a delta set.  The new assert validates that either the parent is scope is a deltaResourceSetScope or we're not attempting to write a child of a delta set (i.e., if the current state is start then we're writing at the top level, not as a child of a delta set.)